File: push/src/main/java/top/trumeet/mipushframework/wizard/permission/UsageStatsPermissionInfo.java
package top.trumeet.mipushframework.wizard.permission;

import android.content.Context;

import androidx.annotation.NonNull;

import com.xiaomi.xmsf.R;

public class UsageStatsPermissionInfo implements PermissionInfo {
    Context context;

    public UsageStatsPermissionInfo(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public PermissionOperator getPermissionOperator() {
        return new UsageStatsPermissionOperator(context);
    }

    @NonNull
    @Override
    public String getPermissionTitle() {
        return context.getString(R.string.wizard_title_stats_permission);
    }

    @NonNull
    @Override
    public String getPermissionDescription() {
        return context.getString(R.string.wizard_title_stats_permission_text);
    }

}

File: push/src/main/java/top/trumeet/mipushframework/wizard/permission/UsageStatsPermissionInfo.kt
package top.trumeet.mipushframework.wizard.permission

import android.content.Context
import com.xiaomi.xmsf.R

class UsageStatsPermissionInfo(private val context: Context) : PermissionInfo {
    override val permissionOperator: PermissionOperator
        get() = UsageStatsPermissionOperator(context)

    override val permissionTitle: String
        get() = context.getString(R.string.wizard_title_stats_permission)

    override val permissionDescription: String
        get() = context.getString(R.string.wizard_title_stats_permission_text)
}
